spec/shopify_transporter/shopify_transporter_spec.rb

```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 require 'shopify_transporter/pipeline/stage'
 RSpec.describe ShopifyTransporter do
-  def tmpfile(content)
+  def tmpfile(content, ext = nil)
```

Comment:
This method's behaviour is a bit awkward.  Maybe just having it take the `content` and `extension`, and having it simply write it to a temp file would be sufficient, with the caller deciding what the format of the content should look like.

Reply:
This method is super super awkward. I was focusing on making it backwards compatible so I wouldn't have to change 100+ lines of test files in the same PR as an unrelated 500 lines of code change. The intent was for a future PR to refactor this into something more sane.

Reply:
Looking at it now, the changes to all those other tests wouldn't be _that_ intensive -- probably just a line or two change per test. Still, I'd prefer to do it in a separate PR since this is already so large. Thoughts?